File: treepat-starter-java/src/main/java/FindMatches.java
import com.github.treepat.expression.TreepatExpression;
import com.github.treepat.target_tree.TargetTree;
import com.github.treepat.target_tree.TargetTreeNode;
import com.github.treepat.target_tree.default_tree.DefaultTargetTree;

import java.util.List;

public class FindMatches {
    public static void main(String[] args) {
        // Treepat Parsing

        TreepatExpression treepatExpression = TreepatExpression.Factory.createFromFile(args[0]);
        // Tree File Parsing
        TargetTree targetTree = DefaultTargetTree.Factory.createFromFile(args[1]);

        List<List<TargetTreeNode>> solutions = targetTree.findMatches(treepatExpression);
        System.out.println(solutions);
        Boolean hasSolutions = targetTree.hasMatch(treepatExpression);
        System.out.println(hasSolutions);
    }
}

File: treepat-starter-kotlin/src/main/kotlin/FindMatches.kt
import com.github.treepat.expression.TreepatExpression
import com.github.treepat.extensions.matchToString
import com.github.treepat.target_tree.default_tree.DefaultTargetTree


fun main(args: Array<String>) {
    // Treepat Parsing
    val treepatExpression = TreepatExpression.createFromFile(args[0])
    // Tree File Parsing
    val targetTree = DefaultTargetTree.createFromFile(args[1])

    val matches = targetTree.findMatches(treepatExpression)
    val printableMatches: List<String>
    if (matches != null) {
        printableMatches = matches.map { it.matchToString() }
    } else {
        printableMatches = listOf("Match not found")
    }
    println(printableMatches.joinToString("\n-\n"))

    val hasSolutions = targetTree.hasMatch(treepatExpression)
    println(hasSolutions)
}

